app/src/main/java/helium314/keyboard/settings/screens/TextCorrectionScreen.kt

```diff
@@ -233,6 +234,11 @@ fun createCorrectionSettings(context: Context) = listOf(
     ) { setting ->
         SwitchPreference(setting, Defaults.PREF_USE_APPS)
     },
+    Setting(
+        context, Settings.PREF_SUGGEST_EMOJIS, R.string.suggest_emojis, R.string.suggest_emojis_summary
+    ) {
+        SwitchPreference(it, Defaults.PREF_SUGGEST_EMOJIS) { KeyboardSwitcher.getInstance().closeDictionaries() }
```

Comment:
Did you try sending an intent with `DictionaryPackConstants.NEW_DICTIONARY_INTENT_ACTION` instead?

Reply:
As far as I understand it should have the same effect, just with a (currently) slightly misleading name.